From a96a921e6b9a685f04dc2b1212a230eb4e34d9ea Mon Sep 17 00:00:00 2001
From: Hauke Mehrtens <hmehrtens@maxlinear.com>
Date: Thu, 10 Jun 2021 15:20:52 +0200
Subject: [PATCH] PONRTSYS-9035: Use dma_alloc_noncoherent()

Instead of calling dma_alloc_attrs() with DMA_ATTR_NON_CONSISTENT use
the wrapper dma_alloc_noncoherent() which does the same.
---
 drivers/net/ethernet/lantiq/cqm/prx300/cqm.c | 12 +++++-------
 1 file changed, 5 insertions(+), 7 deletions(-)

diff --git a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
index bd3f524fe1d3..909e609ec89b 100644
--- a/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
+++ b/drivers/net/ethernet/lantiq/cqm/prx300/cqm.c
@@ -3804,10 +3804,9 @@ static int init_cqm_basic(struct platform_device *pdev)
 	size = (CQM_CPU_POOL_BUF_ALW_NUM + 4) * sizeof(u32);
 	tmp_v = TOT_DMA_HNDL - 1;
 #ifdef CPU_POOL_ALLOWED
-	cqm_ctrl->cpu_rtn_ptr = dma_alloc_attrs(&pdev->dev, size,
+	cqm_ctrl->cpu_rtn_ptr = dma_alloc_noncoherent(&pdev->dev, size,
 						&cqm_ctrl->dma_hndl_p[tmp_v],
-						GFP_KERNEL,
-						DMA_ATTR_NON_CONSISTENT);
+						GFP_KERNEL);
 	dev_info(cqm_ctrl->dev, "cpu rtn buf 0x%p\n", cqm_ctrl->cpu_rtn_ptr);
 	if (!cqm_ctrl->cpu_rtn_ptr) {
 		dev_err(cqm_ctrl->dev, "Error in cpu_rtn_ptr allocation\n");
@@ -3894,10 +3893,9 @@ static int pool_config(struct platform_device *pdev, int p, u32 buf_size,
 	u32 adj_size;
 
 	size = buf_size * buf_num;
-	cqm_ctrl->bm_buf_base[p] = dma_alloc_attrs(&pdev->dev, size,
-			&cqm_ctrl->dma_hndl_p[p],
-			GFP_KERNEL,
-			DMA_ATTR_NON_CONSISTENT);
+	cqm_ctrl->bm_buf_base[p] = dma_alloc_noncoherent(&pdev->dev, size,
+						&cqm_ctrl->dma_hndl_p[p],
+						GFP_KERNEL);
 	dev_dbg(cqm_ctrl->dev, "qos bm buf 0x%p\n", cqm_ctrl->bm_buf_base[p]);
 	if (!cqm_ctrl->bm_buf_base[p]) {
 		dev_err(cqm_ctrl->dev, "Error in pool %d buffer allocation\n",
